blockmanager.go

```diff
@@ -1561,10 +1591,10 @@ func (b *blockManager) handleInvMsg(imsg *invMsg) {
 	}
 }
 
-// limitMap is a helper function for maps that require a maximum limit by
-// evicting a random transaction if adding a new value would cause it to
+// addAndLimitMap is a helper function for maps that require a maximum limit by
+// evicting a random transaction if adding the new value would cause it to
 // overflow the maximum allowed.
-func (b *blockManager) limitMap(m map[chainhash.Hash]struct{}, limit int) {
+func (b *blockManager) addAndLimitMap(m map[chainhash.Hash]struct{}, hash chainhash.Hash, limit int) {
```

Comment:
Since this no longer needs access to the `blockManager`, let's make it a standalone func.  That should've been done back when it was changed so that it didn't directly access the `blockManager` anymore, but I see that it wasn't, so let's do it now.